PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/helpers/IDEHelperImpl.java

```diff
@@ -464,26 +473,66 @@ public void openLinkInBrowser(@NotNull String url) {
 
     private static final Key<String> APP_SERVICE_FILE_ID = new Key<>("APP_SERVICE_FILE_ID");
 
+    @SneakyThrows
     public void openAppServiceFile(final AppServiceFile file, Object context) {
-        final Project project = (Project) context;
-        final FileEditorManager fileEditorManager = FileEditorManager.getInstance(project);
+        final FileEditorManager fileEditorManager = FileEditorManager.getInstance((Project) context);
         final VirtualFile vFile = getOrCreateVirtualFile(file, fileEditorManager);
-        ApplicationManager.getApplication().invokeLater(() -> fileEditorManager.openFile(vFile, true, true), ModalityState.NON_MODAL);
+        final Task.Modal task = new Task.Modal(null, String.format("Opening file %s...", vFile.getName()), true) {
+            @Override
+            public void run(ProgressIndicator indicator) {
+                indicator.setIndeterminate(true);
+                setVirtualFileContent(vFile, file.getContent())
+                    .doOnSubscribe(indicator::start)
+                    .doAfterTerminate(() -> ApplicationManager.getApplication().invokeLater(() -> {
+                        final FileEditor[] editor = fileEditorManager.openFile(vFile, true, true);
+                        if (ArrayUtils.isEmpty(editor)) {
+                            final String message = String.format("Can not open file %s.", vFile.getName());
+                            Messages.showWarningDialog(message, "Open File");
+                        }
+                        indicator.stop();
+                    }, ModalityState.NON_MODAL))
+                    .subscribe();
+            }
+        };
+        ProgressManager.getInstance().run(task);
+    }
+
+    @SneakyThrows
+    private Observable<byte[]> setVirtualFileContent(final VirtualFile vFile, final Observable<byte[]> content) {
+        final OutputStream output = vFile.getOutputStream(null);
```

Comment:
please test against a large file(large than 300M)

Reply:
what will happen if open a large file with intellij? 
we can test, but IMO, if it's Intellij's problem if it can not open it.

Reply:
We may add a file size limitation here? For large file, only stream download is enabled.